pkg/terraform/terraform.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package Terraform defines an SDK for running Terraform commands using
+// the Terraform CLI binary.
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+
+	"github.com/abcxyz/guardian/pkg/child"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+)
+
+var _ Terraform = (*TerraformClient)(nil)
+
+// Terraform is the interface for working with the Terraform CLI.
+type Terraform interface {
+	Init(ctx context.Context, workingDir string, args ...string) ([]byte, []byte, int, error)
+	Validate(ctx context.Context, workingDir string, args ...string) ([]byte, []byte, int, error)
+	Plan(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	Apply(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	Show(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	GetEntrypointDirectories(rootDir string) ([]string, error)
+}
+
+// TerraformClient implements the Terraform interface.
+type TerraformClient struct{}
+
+// NewTerraformClient creates a new Terraform client.
+func NewTerraformClient() *TerraformClient {
+	return &TerraformClient{}
+}
+
+// Init runs the terraform init command.
+func (t *TerraformClient) Init(ctx context.Context, workingDir string, args ...string) ([]byte, []byte, int, error) {
```

Comment:
These return types are strange. I assume it's stdout, stderr, exit code, and any error?

It might be better to return a `(struct, error)`, and the structure holds stdout, stderr, and the return code. (throughout)

Reply:
Should I adopt this in `child.go` instead of here?

Reply:
Yea probably